ci_framework/roles/os_must_gather/tasks/main.yml

```diff
@@ -14,6 +14,22 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+- name: Construct project change list
+  ansible.builtin.set_fact:
+    zuul_change_list: "{{ zuul_change_list | default([]) + [item.project.short_name] }}"
+    cacheable: true
+  with_items: "{{ zuul['items'] }}"
+  when:
+    - zuul is defined
+    - "'change_url' in item"
+
+- name: Build openstack-must-gather image
+  ansible.builtin.import_tasks: build_openstack-must-gather_image.yml
+  when:
+    - zuul_change_list is defined
+    #- "'openstack-must-gather' in zuul_change_list"
+    - registry_login_enabled | default('false') | bool
```

Comment:
Maybe, if we use a local registry, I think that this variable is not set. It is better to control when or not to push with a control var, inside the playbook?